vyper/semantics/analysis/module.py

```diff
@@ -265,8 +265,7 @@ def visit_FunctionDef(self, node):
         func = ContractFunctionT.from_FunctionDef(node)
 
         try:
-            # TODO sketchy elision of namespace validation
```

Comment:
Why was it like this?

Reply:
Based on the only failing test, I believe it's to accommodate custom `send` methods like this example which used to be valid, but no longer so with this change:
```
# Test custom send method
@internal
def send(a: address, w: uint256):
    send(0xde0B295669a9FD93d5F28D9Ec85E40f4cb697BAe, 1)

@external
@payable
def foo():
    self.send(msg.sender, msg.value)
```

Reply:
I did some digging and it's [related](https://github.com/vyperlang/vyper/issues/1068) to the [ERC-777](https://eips.ethereum.org/EIPS/eip-777) interface having a `send` function. Since the prevailing consensus is to recommend against using ERC-777 at all (see OZ [deprecating](https://github.com/OpenZeppelin/openzeppelin-contracts/issues/2620) their implementation), I think it is now appropriate to remove this workaround.